packages/sitecore-jss/src/graphql-request-client.test.ts

```diff
@@ -405,4 +406,40 @@ describe('GraphQLRequestClient', () => {
       }
     });
   });
+
+  describe('DefaultRetryStrategy', () => {
+    const mockClientError = new ClientError(
+      {
+        data: undefined,
+        errors: [{ message: 'GaphqlError' }],
+        extensions: undefined,
+        status: 502,
+      },
+      {
+        query: 'query',
+      }
+    );
+
+    const retryStrategy = new DefaultRetryStrategy([429, 503]);
+
+    it('should return true from shouldRetry when conditions are met', () => {
+      mockClientError.response.status = 503;
+
+      const shouldRetry = retryStrategy.shouldRetry(mockClientError, 1, 3);
+      expect(shouldRetry).to.equal(true);
+    });
+
+    it('should return false from shouldRetry method when conditions are not met', () => {
+      mockClientError.response.status = 503;
+
+      const shouldRetry = retryStrategy.shouldRetry(mockClientError, 2, 1);
+      expect(shouldRetry).to.equal(false);
+    });
+
+    it('should return delay using exponential backoff when Retry-After header is not present', () => {
+      const delay = retryStrategy.getDelay(mockClientError, 3);
+      const expectedDelay = Math.pow(retryStrategy['factor'], 3 - 1) * 1000;
+      expect(delay).to.equal(expectedDelay);
+    });
+  });
```

Comment:
There are a few more tests we could add here for better coverage, for example:
- shouldRetry returns false when retries is 0
- shouldRetry returns false when attempt exceeds retries
- shouldRetry returns false when error response status undefined
- getDelay uses custom exponential factor